// Copyright 2020 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator.proto";
import "mediapipe/framework/formats/classification.proto";
import "mediapipe/framework/formats/landmark.proto";
import "mediapipe/framework/formats/matrix_data.proto";
import "mediapipe/framework/formats/time_series_header.proto";

message ConstantSidePacketCalculatorOptions {
  extend CalculatorOptions {
    optional ConstantSidePacketCalculatorOptions ext = 291214597;
  }

  message StringVector {
    repeated string string_value = 1;
  }
  message ConstantSidePacket {
    oneof value {
      int32 int_value = 1;
      uint64 uint64_value = 5;
      int64 int64_value = 11;
      float float_value = 2;
      double double_value = 9;
      bool bool_value = 3;
      string string_value = 4;
      ClassificationList classification_list_value = 6;
      LandmarkList landmark_list_value = 7;
      TimeSeriesHeader time_series_header_value = 10;
      MatrixData matrix_data_value = 12;
      StringVector string_vector_value = 13;
    }
  }

  repeated ConstantSidePacket packet = 1;
}
